### PR TITLE
[IMPROVEMENT] Remove RPC URL metrics

### LINKED ISSUE #5315: [PRIVACY] Metrics events or properties updates
- Remove RPC url property
- Remove blockexplorer url property

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_

Stop logging the RPC URL in our metrics events

_2. What is the improvement/solution?_

Remove the relevant parameter from the following events,

- Network Added
- Network Switched
- Network Requested
- Network Request Rejected

**Screenshots/Recordings**

_Not applicable_

**Issue**

Progresses https://github.com/MetaMask/metamask-mobile/issues/5302

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/UI/NetworkList/index.js [hunk 1]
```diff
@@ -237,7 +237,6 @@ export class NetworkList extends PureComponent {
     NetworkController.setRpcTarget(rpcUrl, chainId, ticker, nickname);
 
     AnalyticsV2.trackEvent(AnalyticsV2.ANALYTICS_EVENTS.NETWORK_SWITCHED, {
-      rpc_url: rpcUrl,
       chain_id: chainId,
       source: 'Settings',
       symbol: ticker,
```
Comment (blackdevelopa) [inline app/components/UI/NetworkList/index.js:242]: Should we also remove `block_explorer_url` in line 243?

File: app/components/UI/NetworkModal/index.tsx [hunk 2]
```diff
@@ -168,7 +166,6 @@ const NetworkModals = (props: NetworkProps) => {
       );
 
       const analyticsParamsAdd = {
-        rpc_url: sanitizedUrl,
         chain_id: decimalChainId,
         source: 'Popular network list',
         symbol: ticker,
```
Comment (blackdevelopa) [inline app/components/UI/NetworkModal/index.tsx:171]: Nitpicking: Should we also remove `blockExplorerUrl` in line 172?

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js [hunk 3]
```diff
@@ -484,7 +484,6 @@ class NetworkSettings extends PureComponent {
       );
 
       const analyticsParamsAdd = {
-        rpc_url: formattedHref,
         chain_id: decimalChainId,
         source: 'Custom network form',
         symbol: ticker,
```
Comment (blackdevelopa) [inline app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js:489]: Same here in line 490

File: app/core/RPCMethods/wallet_addEthereumChain.js [hunk 4]
```diff
@@ -125,7 +125,6 @@ const wallet_addEthereumChain = async ({
     }
 
     const analyticsParams = {
-      rpc_url: existingNetwork?.rpcUrl,
       chain_id: _chainId,
       source: 'Custom Network API',
       symbol: existingNetwork?.ticker,
```
Comment (blackdevelopa) [inline app/core/RPCMethods/wallet_addEthereumChain.js:130]: Line 131?

File: app/core/RPCMethods/wallet_switchEthereumChain.js [hunk 5]
```diff
@@ -81,7 +81,6 @@ const wallet_switchEthereumChain = async ({
       };
       analyticsParams = {
         ...analyticsParams,
-        rpc_url: existingNetworkRPC?.rpcUrl,
         symbol: existingNetworkRPC?.ticker,
         block_explorer_url: existingNetworkRPC?.blockExplorerUrl,
```
Comment (blackdevelopa) [inline app/core/RPCMethods/wallet_switchEthereumChain.js:85]: Line 85?

**Review by blackdevelopa — COMMENTED** [review]
LGTM. My only comment is about not removing `blockExplorerUrl` as well. Does it matter less?

File: app/components/UI/NetworkList/index.js (hunk 1 above)
Comment (gantunesr) [reply app/components/UI/NetworkList/index.js:236]: [c2928eb](https://github.com/MetaMask/metamask-mobile/pull/5307/commits/c2928ebbca6de0f45dca94812d9ff15cac073785)

File: app/core/RPCMethods/wallet_switchEthereumChain.js (hunk 5 above)
Comment (gantunesr) [reply app/core/RPCMethods/wallet_switchEthereumChain.js]: [c2928eb](https://github.com/MetaMask/metamask-mobile/pull/5307/commits/c2928ebbca6de0f45dca94812d9ff15cac073785)

File: app/core/RPCMethods/wallet_addEthereumChain.js (hunk 4 above)
Comment (gantunesr) [reply app/core/RPCMethods/wallet_addEthereumChain.js:130]: [c2928eb](https://github.com/MetaMask/metamask-mobile/pull/5307/commits/c2928ebbca6de0f45dca94812d9ff15cac073785)

File: app/components/UI/NetworkModal/index.tsx (hunk 2 above)
Comment (gantunesr) [reply app/components/UI/NetworkModal/index.tsx:171]: [c2928eb](https://github.com/MetaMask/metamask-mobile/pull/5307/commits/c2928ebbca6de0f45dca94812d9ff15cac073785)

**Review by blackdevelopa — APPROVED** [review]
Looks good to me. Thank you, @gantunesr

**Pull request merged**